src/components/common/OfficeAddresses.vue

```diff
@@ -278,185 +270,135 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
    */
-  @Emit('update:addresses')
-  private emitAddresses (): object {
-    let deliveryAddressFinal = {} as AddressIF
-    let mailingAddressFinal = {} as AddressIF
-    let recDeliveryAddressFinal = {} as AddressIF
-    let recMailingAddressFinal = {} as AddressIF
-
-    if (this.registeredAddress) {
-      deliveryAddressFinal = { ...this.deliveryAddress }
-      mailingAddressFinal = { ...this.mailingAddress }
-
-      // if the address has changed from the original, set action flag
-      this.addressModified(this.deliveryAddress, this.registeredAddress.deliveryAddress)
-        ? this.addAction(deliveryAddressFinal, Actions.ADDRESSCHANGED)
-        : this.removeAction(deliveryAddressFinal, Actions.ADDRESSCHANGED)
-
-      // if the address has changed from the original, set action flag
-      this.addressModified(this.mailingAddress, this.registeredAddress.mailingAddress)
-        ? this.addAction(mailingAddressFinal, Actions.ADDRESSCHANGED)
-        : this.removeAction(mailingAddressFinal, Actions.ADDRESSCHANGED)
+  private async fetchAddresses (): Promise<void> {
+    if (this.entityIncNo && this.asOfDate) {
+      const url = `businesses/${this.entityIncNo}/addresses?date=${this.asOfDate}`
+      console.log('*** fetching addresses...')
+      await axios.get(url)
+        .then(response => {
+          const registeredOffice = response?.data?.registeredOffice
+          const recordsOffice = response?.data?.recordsOffice
+          // registered office is mandatory
+          if (registeredOffice) {
+            this.original.registeredOffice = {
+              mailingAddress: { ...registeredOffice.mailingAddress, actions: [] },
+              deliveryAddress: { ...registeredOffice.deliveryAddress, actions: [] }
+            }
+          } else {
+            throw new Error('Missing registered office address in response = ' + response)
+          }
+
+          // records office is mandatory only for BCOMP
+          if (recordsOffice) {
+            this.original.recordsOffice = {
+              mailingAddress: { ...recordsOffice.mailingAddress, actions: [] },
+              deliveryAddress: { ...recordsOffice.deliveryAddress, actions: [] }
+            }
+          } else if (this.isBComp) {
+            throw new Error('Missing records office address in response = ' + response)
+          }
+        })
+        .catch(error => {
+          // eslint-disable-next-line no-console
+          console.log('fetchAddresses() error =', error)
+
+          // emit event to parent to display Fetch Error Dialog
+          this.$root.$emit('fetch-error-event')
+        })
     }
+  }
 
-    if (this.recordsAddress) {
-      recDeliveryAddressFinal = { ...this.recDeliveryAddress }
-      recMailingAddressFinal = { ...this.recMailingAddress }
+  /** Called when As Of Date prop is set or changes. */
+  @Watch('asOfDate', { immediate: true })
+  private async onAsOfDateChanged (date: string): Promise<void> {
```

Comment:
This watcher is the trigger for fetching the original addresses from the API. It all depends on this.